src/upload/solana.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::{anyhow, Result};
+use retry::delay::Exponential;
+use retry::retry;
+use solana_client::rpc_client::RpcClient;
+use solana_json::instruction::SetValueArgs;
+use solana_sdk::{signature::Keypair, signer::Signer, transaction::Transaction};
+use std::fs::File;
+
+use crate::update::{parse_keypair, parse_solana_config};
+
+pub struct UploadSolanaArgs {
+    pub client: RpcClient,
+    pub keypair: Option<String>,
+    pub json_file: String,
+    pub rate_limit: usize,
+    pub retries: u8,
```

Comment:
`rate_limit` and `retries` aren't used at all. Is the idea to add them to the API for a future async refactor that will use them?

Reply:
No, my bad. I forgot to add their usage.